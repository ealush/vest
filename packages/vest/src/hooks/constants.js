/**
 *  @type {String}  Error message to display when a hook was called outside of context.
 */
export const ERROR_HOOK_CALLED_OUTSIDE =
  "hook called outside of a running suite.";
