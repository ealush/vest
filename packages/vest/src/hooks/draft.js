import context from 'ctx';
import { ERROR_HOOK_CALLED_OUTSIDE } from 'hookErrors';
import produce from 'produce';
import throwError from 'throwError';

/**
 * @returns {Object} Current output object.
 */
const draft = () => {
  const { stateRef } = context.use();

  if (!stateRef) {
    throwError('draft ' + ERROR_HOOK_CALLED_OUTSIDE);
    return;
  }

  return produce(/*isDraft:*/ true);
};

export default draft;
