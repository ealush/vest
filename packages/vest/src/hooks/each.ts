import isFunction from 'isFunction';
import throwError from 'throwError';

import { IsolateTypes } from 'IsolateTypes';
import { isolate } from 'isolate';

export default function each<T>(
  list: T[],
  callback: (arg: T, index: number) => void
): void {
  if (!isFunction(callback)) {
    throwError('callback must be a function');
  }

  isolate({ type: IsolateTypes.EACH }, () => {
    list.forEach((arg, index) => {
      callback(arg, index);
    });
  });
}
