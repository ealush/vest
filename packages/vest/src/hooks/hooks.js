export { only, skip } from 'exclusive';
export { default as warn } from 'warn';
export { default as group } from 'group';
