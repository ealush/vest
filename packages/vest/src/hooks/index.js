export { default as draft } from './draft';
export { only, skip } from './exclusive';
export { default as warn } from './warn';
export { default as get } from './get';
export { default as group } from './group';
