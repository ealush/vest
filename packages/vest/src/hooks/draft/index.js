import context from '../../core/context';
import produce from '../../core/produce';
import throwError from '../../lib/throwError';
import { ERROR_HOOK_CALLED_OUTSIDE } from '../constants';

/**
 * @returns {Object} Current output object.
 */
const draft = () => {
  const { stateRef } = context.use();

  if (stateRef === undefined) {
    throwError('draft ' + ERROR_HOOK_CALLED_OUTSIDE);
    return;
  }

  return produce({ draft: true });
};

export default draft;
