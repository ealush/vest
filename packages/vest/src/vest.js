import create from 'createSuite';
import enforce from 'enforce';
import { only, skip, warn, group } from 'hooks';
import test from 'test';

const VERSION = __LIB_VERSION__;

export default {
  VERSION,
  create,
  enforce,
  group,
  only,
  skip,
  test,
  warn,
};
