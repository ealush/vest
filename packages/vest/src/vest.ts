import { enforce } from 'n4s';

import create from 'create';
import context from 'ctx';
import each from 'each';
import { only, skip } from 'exclusive';
import group from 'group';
import include from 'include';
import { eager } from 'mode';
import omitWhen from 'omitWhen';
import optional from 'optionalTests';
import skipWhen from 'skipWhen';
import { test } from 'test';
import warn from 'warn';

const VERSION = __LIB_VERSION__;
export {
  test,
  create,
  each,
  only,
  skip,
  warn,
  group,
  optional,
  skipWhen,
  omitWhen,
  enforce,
  VERSION,
  context,
  include,
  eager,
};
