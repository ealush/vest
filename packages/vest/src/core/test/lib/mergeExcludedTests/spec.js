import _ from 'lodash';
import { dummyTest } from '../../../../../testUtils/testDummy';
import { group } from '../../../../hooks';
import vest from '../../../../index';

const SUITE_NAME = 'suite_1';
const FIELD_NAME_1 = 'field_1';
const FIELD_NAME_2 = 'field_2';
const GROUP_NAME_1 = 'group_1';
const GROUP_NAME_2 = 'group_2';

describe('When previous state exists', () => {
  describe('When no currently skipped fields or groups', () => {
    describe('When skipped field exists in previous state', () => {
      let validate, prevRes, res;

      const validation = () =>
        vest.create(SUITE_NAME, skip => {
          vest.skip(skip);
          dummyTest.failingWarning(FIELD_NAME_1, 'warning message');

          group(GROUP_NAME_1, () => {
            dummyTest.failing(FIELD_NAME_2, 'error message');
          });
        });
      beforeEach(() => {
        validate = validation();
        prevRes = _.cloneDeep(validate());
      });

      describe('Warning field', () => {
        it('Should copy prev field state over', () => {
          res = validate(/* skip */ FIELD_NAME_1);

          expect(res.tests[FIELD_NAME_1]).toEqual(prevRes.tests[FIELD_NAME_1]);
        });
      });
      describe('Error field', () => {
        it('Should copy prev field state over', () => {
          res = validate(/* skip */ FIELD_NAME_2);

          expect(res.tests[FIELD_NAME_2]).toEqual(prevRes.tests[FIELD_NAME_2]);
        });
      });
    });
  });

  describe('When skipped group exists in previous state', () => {
    let res, validate;

    const validation = () =>
      vest.create(SUITE_NAME, skip => {
        vest.skip.group(skip);

        dummyTest.failing('f1', 'f1_msg');
        dummyTest.failing('f2', 'f2_msg');
        dummyTest.failing('f3', 'f3_msg');
        dummyTest.failingAsync('f3', { statement: 'f3_async_msg' });

        group(GROUP_NAME_1, () => {
          dummyTest.failing('f1', 'f1_group_1_msg');
          dummyTest.failingWarning('f2', 'f2_group_1_msg');
          dummyTest.failing('f3', 'f3_group_1_msg');
          dummyTest.failingAsync('f3', {
            statement: 'f3_async_group_1_msg',
          });
          dummyTest.failing('f4', 'f4_group_1_msg');
          dummyTest.passing('f5', 'f5_group_1_msg');
        });

        group(GROUP_NAME_2, () => {
          dummyTest.failing('f1', 'f1_group_2_msg');
          dummyTest.failingAsync('f3', {
            statement: 'f3_async_group_2_msg',
          });
          dummyTest.failing('f4', 'f4_group_2_msg');
          dummyTest.failing('f5', 'f5_group_2_msg');
          dummyTest.failingWarning('f6', 'f6_group_2_msg');
          dummyTest.passing('f7', 'f7_group_2_msg');
        });
      });

    beforeEach(() => {
      validate = validation();
    });

    test('sanity: skipped tests are not in initial state', () =>
      new Promise(done => {
        res = _.cloneDeep(validate([GROUP_NAME_1, GROUP_NAME_2]));
        expect(res.tests).toHaveProperty('f1');
        expect(res.tests).toHaveProperty('f2');
        expect(res.tests).toHaveProperty('f3');
        expect(res.errorCount).toBe(3);
        expect(res.warnCount).toBe(0);
        expect(res.testCount).toBe(4);
        expect(res.tests.f1.errorCount).toBe(1);
        expect(res.tests.f1.warnCount).toBe(0);
        expect(res.tests.f1.errorCount).toBe(1);
        expect(res.tests.f2.warnCount).toBe(0);
        expect(res.tests.f3.errorCount).toBe(1);
        expect(res.tests).not.toHaveProperty('f4');
        expect(res.tests).not.toHaveProperty('f5');
        expect(res.tests).not.toHaveProperty('f6');
        expect(res.tests).not.toHaveProperty('f7');
        expect(res).toMatchSnapshot();
        setTimeout(() => {
          res = validate.get();
          expect(res.errorCount).toBe(4);
          expect(res.tests.f3.errorCount).toBe(2);
          done();
        });
      }));

    test('Skipping after initial run: Should copy group state over', () =>
      new Promise(done => {
        // skipping group 2
        res = _.cloneDeep(validate(GROUP_NAME_2));
        expect(res.errorCount).toBe(6);
        expect(res.warnCount).toBe(1);
        expect(res.testCount).toBe(10);
        expect(res.tests.f1.errorCount).toBe(2);
        expect(res.tests.f1.warnCount).toBe(0);
        expect(res.tests.f2.errorCount).toBe(1);
        expect(res.tests.f2.warnCount).toBe(1);
        expect(res.tests.f3.errorCount).toBe(2);
        expect(res.tests.f4.errorCount).toBe(1);
        expect(res.tests.f4.warnCount).toBe(0);
        expect(res.tests.f5.errorCount).toBe(0);
        expect(res.tests.f5.warnCount).toBe(0);
        expect(res).toMatchSnapshot();
        setTimeout(() => {
          res = validate.get();
          // both outer and inner async
          expect(res.tests.f3.errorCount).toBe(4);
          done();
        });
      }));

    test('Skipping after second run: Should copy group state over', () =>
      new Promise(done => {
        _.cloneDeep(validate(GROUP_NAME_2));

        res = _.cloneDeep(validate(GROUP_NAME_1));

        expect(res.errorCount).toBe(9);
        expect(res.warnCount).toBe(2);
        expect(res.testCount).toBe(16);

        expect(res.tests.f2.errorCount).toBe(1);
        expect(res.tests.f2.warnCount).toBe(1);

        expect(res.tests.f3.errorCount).toBe(2);
        expect(res.tests.f3.warnCount).toBe(0);

        expect(res.tests.f4.errorCount).toBe(2);
        expect(res.tests.f4.warnCount).toBe(0);

        expect(res.tests.f5.errorCount).toBe(1);
        expect(res.tests.f5.warnCount).toBe(0);

        expect(res.tests.f6.errorCount).toBe(0);
        expect(res.tests.f6.warnCount).toBe(1);

        expect(res.tests.f7.errorCount).toBe(0);
        expect(res.tests.f7.warnCount).toBe(0);
        expect(res).toMatchSnapshot();

        setTimeout(() => {
          res = validate.get();
          expect(res.tests.f3.errorCount).toBe(5);
          expect(res).toMatchSnapshot();
          done();
        });
      }));
  });
});
