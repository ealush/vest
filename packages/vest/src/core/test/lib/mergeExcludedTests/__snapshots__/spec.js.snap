// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`module: skipped export: mergeExcludedTests When previous state exists When no currently skipped fields or groups When skipped field exists in previous state Error field Should copy prev field state over 1`] = `
Object {
  "doneCallbacks": Array [],
  "fieldCallbacks": Object {},
  "groups": Object {
    "group_1": Object {},
  },
  "lagging": Array [],
  "name": "suite_1",
  "pending": Array [],
  "suiteId": "suite_1",
  "testObjects": Array [
    VestTest {
      "failed": true,
      "fieldName": "field_1",
      "id": "6",
      "isWarning": true,
      "statement": "warning message",
      "suiteId": "suite_1",
      "testFn": [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "type": "throw",
            "value": [Error],
          },
        ],
      },
    },
    VestTest {
      "failed": true,
      "fieldName": "field_2",
      "groupName": "group_1",
      "id": "5",
      "isWarning": false,
      "statement": "error message",
      "suiteId": "suite_1",
      "testFn": [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "type": "throw",
            "value": [Error],
          },
        ],
      },
    },
  ],
  "tests": Object {},
}
`;

exports[`module: skipped export: mergeExcludedTests When previous state exists When no currently skipped fields or groups When skipped field exists in previous state Warning field Should copy prev field state over 1`] = `
Object {
  "doneCallbacks": Array [],
  "fieldCallbacks": Object {},
  "groups": Object {
    "group_1": Object {},
  },
  "lagging": Array [],
  "name": "suite_1",
  "pending": Array [],
  "suiteId": "suite_1",
  "testObjects": Array [
    VestTest {
      "failed": true,
      "fieldName": "field_2",
      "groupName": "group_1",
      "id": "3",
      "isWarning": false,
      "statement": "error message",
      "suiteId": "suite_1",
      "testFn": [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "type": "throw",
            "value": [Error],
          },
        ],
      },
    },
    VestTest {
      "failed": true,
      "fieldName": "field_1",
      "id": "0",
      "isWarning": true,
      "statement": "warning message",
      "suiteId": "suite_1",
      "testFn": [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "type": "throw",
            "value": [Error],
          },
        ],
      },
    },
  ],
  "tests": Object {},
}
`;

exports[`module: skipped export: mergeExcludedTests When previous state exists When skipped group exists in previous state Skipping after initial run: Should copy group state over 1`] = `
Object {
  "done": [Function],
  "errorCount": 6,
  "getErrors": [Function],
  "getErrorsByGroup": [Function],
  "getWarnings": [Function],
  "getWarningsByGroup": [Function],
  "groups": Object {
    "group_1": Object {
      "f1": Object {
        "errorCount": 1,
        "errors": Array [
          "f1_group_1_msg",
        ],
        "testCount": 1,
        "warnCount": 0,
      },
      "f2": Object {
        "errorCount": 0,
        "testCount": 1,
        "warnCount": 1,
        "warnings": Array [
          "f2_group_1_msg",
        ],
      },
      "f3": Object {
        "errorCount": 1,
        "errors": Array [
          "f3_group_1_msg",
        ],
        "testCount": 2,
        "warnCount": 0,
      },
      "f4": Object {
        "errorCount": 1,
        "errors": Array [
          "f4_group_1_msg",
        ],
        "testCount": 1,
        "warnCount": 0,
      },
      "f5": Object {
        "errorCount": 0,
        "testCount": 1,
        "warnCount": 0,
      },
    },
  },
  "hasErrors": [Function],
  "hasErrorsByGroup": [Function],
  "hasWarnings": [Function],
  "hasWarningsByGroup": [Function],
  "name": "suite_1",
  "testCount": 10,
  "tests": Object {
    "f1": Object {
      "errorCount": 2,
      "errors": Array [
        "f1_msg",
        "f1_group_1_msg",
      ],
      "testCount": 2,
      "warnCount": 0,
    },
    "f2": Object {
      "errorCount": 1,
      "errors": Array [
        "f2_msg",
      ],
      "testCount": 2,
      "warnCount": 1,
      "warnings": Array [
        "f2_group_1_msg",
      ],
    },
    "f3": Object {
      "errorCount": 2,
      "errors": Array [
        "f3_msg",
        "f3_group_1_msg",
      ],
      "testCount": 4,
      "warnCount": 0,
    },
    "f4": Object {
      "errorCount": 1,
      "errors": Array [
        "f4_group_1_msg",
      ],
      "testCount": 1,
      "warnCount": 0,
    },
    "f5": Object {
      "errorCount": 0,
      "testCount": 1,
      "warnCount": 0,
    },
  },
  "warnCount": 1,
}
`;

exports[`module: skipped export: mergeExcludedTests When previous state exists When skipped group exists in previous state Skipping after second run: Should copy group state over 1`] = `
Object {
  "done": [Function],
  "errorCount": 9,
  "getErrors": [Function],
  "getErrorsByGroup": [Function],
  "getWarnings": [Function],
  "getWarningsByGroup": [Function],
  "groups": Object {
    "group_1": Object {
      "f1": Object {
        "errorCount": 1,
        "errors": Array [
          "f1_group_1_msg",
        ],
        "testCount": 1,
        "warnCount": 0,
      },
      "f2": Object {
        "errorCount": 0,
        "testCount": 1,
        "warnCount": 1,
        "warnings": Array [
          "f2_group_1_msg",
        ],
      },
      "f3": Object {
        "errorCount": 1,
        "errors": Array [
          "f3_group_1_msg",
        ],
        "testCount": 2,
        "warnCount": 0,
      },
      "f4": Object {
        "errorCount": 1,
        "errors": Array [
          "f4_group_1_msg",
        ],
        "testCount": 1,
        "warnCount": 0,
      },
      "f5": Object {
        "errorCount": 0,
        "testCount": 1,
        "warnCount": 0,
      },
    },
    "group_2": Object {
      "f1": Object {
        "errorCount": 1,
        "errors": Array [
          "f1_group_2_msg",
        ],
        "testCount": 1,
        "warnCount": 0,
      },
      "f3": Object {
        "errorCount": 0,
        "testCount": 1,
        "warnCount": 0,
      },
      "f4": Object {
        "errorCount": 1,
        "errors": Array [
          "f4_group_2_msg",
        ],
        "testCount": 1,
        "warnCount": 0,
      },
      "f5": Object {
        "errorCount": 1,
        "errors": Array [
          "f5_group_2_msg",
        ],
        "testCount": 1,
        "warnCount": 0,
      },
      "f6": Object {
        "errorCount": 0,
        "testCount": 1,
        "warnCount": 1,
        "warnings": Array [
          "f6_group_2_msg",
        ],
      },
      "f7": Object {
        "errorCount": 0,
        "testCount": 1,
        "warnCount": 0,
      },
    },
  },
  "hasErrors": [Function],
  "hasErrorsByGroup": [Function],
  "hasWarnings": [Function],
  "hasWarningsByGroup": [Function],
  "name": "suite_1",
  "testCount": 16,
  "tests": Object {
    "f1": Object {
      "errorCount": 3,
      "errors": Array [
        "f1_msg",
        "f1_group_2_msg",
        "f1_group_1_msg",
      ],
      "testCount": 3,
      "warnCount": 0,
    },
    "f2": Object {
      "errorCount": 1,
      "errors": Array [
        "f2_msg",
      ],
      "testCount": 2,
      "warnCount": 1,
      "warnings": Array [
        "f2_group_1_msg",
      ],
    },
    "f3": Object {
      "errorCount": 2,
      "errors": Array [
        "f3_msg",
        "f3_group_1_msg",
      ],
      "testCount": 5,
      "warnCount": 0,
    },
    "f4": Object {
      "errorCount": 2,
      "errors": Array [
        "f4_group_2_msg",
        "f4_group_1_msg",
      ],
      "testCount": 2,
      "warnCount": 0,
    },
    "f5": Object {
      "errorCount": 1,
      "errors": Array [
        "f5_group_2_msg",
      ],
      "testCount": 2,
      "warnCount": 0,
    },
    "f6": Object {
      "errorCount": 0,
      "testCount": 1,
      "warnCount": 1,
      "warnings": Array [
        "f6_group_2_msg",
      ],
    },
    "f7": Object {
      "errorCount": 0,
      "testCount": 1,
      "warnCount": 0,
    },
  },
  "warnCount": 2,
}
`;

exports[`module: skipped export: mergeExcludedTests When previous state exists When skipped group exists in previous state Skipping after second run: Should copy group state over 2`] = `
Object {
  "done": [Function],
  "errorCount": 12,
  "getErrors": [Function],
  "getErrorsByGroup": [Function],
  "getWarnings": [Function],
  "getWarningsByGroup": [Function],
  "groups": Object {
    "group_1": Object {
      "f1": Object {
        "errorCount": 1,
        "errors": Array [
          "f1_group_1_msg",
        ],
        "testCount": 1,
        "warnCount": 0,
      },
      "f2": Object {
        "errorCount": 0,
        "testCount": 1,
        "warnCount": 1,
        "warnings": Array [
          "f2_group_1_msg",
        ],
      },
      "f3": Object {
        "errorCount": 2,
        "errors": Array [
          "f3_group_1_msg",
          "f3_async_group_1_msg",
        ],
        "testCount": 2,
        "warnCount": 0,
      },
      "f4": Object {
        "errorCount": 1,
        "errors": Array [
          "f4_group_1_msg",
        ],
        "testCount": 1,
        "warnCount": 0,
      },
      "f5": Object {
        "errorCount": 0,
        "testCount": 1,
        "warnCount": 0,
      },
    },
    "group_2": Object {
      "f1": Object {
        "errorCount": 1,
        "errors": Array [
          "f1_group_2_msg",
        ],
        "testCount": 1,
        "warnCount": 0,
      },
      "f3": Object {
        "errorCount": 1,
        "errors": Array [
          "f3_async_group_2_msg",
        ],
        "testCount": 1,
        "warnCount": 0,
      },
      "f4": Object {
        "errorCount": 1,
        "errors": Array [
          "f4_group_2_msg",
        ],
        "testCount": 1,
        "warnCount": 0,
      },
      "f5": Object {
        "errorCount": 1,
        "errors": Array [
          "f5_group_2_msg",
        ],
        "testCount": 1,
        "warnCount": 0,
      },
      "f6": Object {
        "errorCount": 0,
        "testCount": 1,
        "warnCount": 1,
        "warnings": Array [
          "f6_group_2_msg",
        ],
      },
      "f7": Object {
        "errorCount": 0,
        "testCount": 1,
        "warnCount": 0,
      },
    },
  },
  "hasErrors": [Function],
  "hasErrorsByGroup": [Function],
  "hasWarnings": [Function],
  "hasWarningsByGroup": [Function],
  "name": "suite_1",
  "testCount": 16,
  "tests": Object {
    "f1": Object {
      "errorCount": 3,
      "errors": Array [
        "f1_msg",
        "f1_group_2_msg",
        "f1_group_1_msg",
      ],
      "testCount": 3,
      "warnCount": 0,
    },
    "f2": Object {
      "errorCount": 1,
      "errors": Array [
        "f2_msg",
      ],
      "testCount": 2,
      "warnCount": 1,
      "warnings": Array [
        "f2_group_1_msg",
      ],
    },
    "f3": Object {
      "errorCount": 5,
      "errors": Array [
        "f3_msg",
        "f3_async_msg",
        "f3_async_group_2_msg",
        "f3_group_1_msg",
        "f3_async_group_1_msg",
      ],
      "testCount": 5,
      "warnCount": 0,
    },
    "f4": Object {
      "errorCount": 2,
      "errors": Array [
        "f4_group_2_msg",
        "f4_group_1_msg",
      ],
      "testCount": 2,
      "warnCount": 0,
    },
    "f5": Object {
      "errorCount": 1,
      "errors": Array [
        "f5_group_2_msg",
      ],
      "testCount": 2,
      "warnCount": 0,
    },
    "f6": Object {
      "errorCount": 0,
      "testCount": 1,
      "warnCount": 1,
      "warnings": Array [
        "f6_group_2_msg",
      ],
    },
    "f7": Object {
      "errorCount": 0,
      "testCount": 1,
      "warnCount": 0,
    },
  },
  "warnCount": 2,
}
`;

exports[`module: skipped export: mergeExcludedTests When previous state exists When skipped group exists in previous state sanity: skipped tests are not in initial state 1`] = `
Object {
  "done": [Function],
  "errorCount": 3,
  "getErrors": [Function],
  "getErrorsByGroup": [Function],
  "getWarnings": [Function],
  "getWarningsByGroup": [Function],
  "groups": Object {},
  "hasErrors": [Function],
  "hasErrorsByGroup": [Function],
  "hasWarnings": [Function],
  "hasWarningsByGroup": [Function],
  "name": "suite_1",
  "testCount": 4,
  "tests": Object {
    "f1": Object {
      "errorCount": 1,
      "errors": Array [
        "f1_msg",
      ],
      "testCount": 1,
      "warnCount": 0,
    },
    "f2": Object {
      "errorCount": 1,
      "errors": Array [
        "f2_msg",
      ],
      "testCount": 1,
      "warnCount": 0,
    },
    "f3": Object {
      "errorCount": 1,
      "errors": Array [
        "f3_msg",
      ],
      "testCount": 2,
      "warnCount": 0,
    },
  },
  "warnCount": 0,
}
`;
